fio.spec

```diff
@@ -1,17 +1,21 @@
+%global shortcommit %(c=%{commit};echo ${c:0:7})
+
 Name:		fio
-Version:	3.20
-Release:	1%{?dist}
+Version:	3.36
+Release:	1%{?commit:.g%{shortcommit}}%{?dist}
 Summary:	Multithreaded IO generation tool
 
 Group:		Applications/System
 License:	GPLv2
 URL:		http://git.kernel.dk/?p=fio.git;a=summary
 Source:		http://brick.kernel.dk/snaps/%{name}-%{version}.tar.bz2
+Source1:	pathfix.py
```

Comment:
Where did this come from?

Reply:
> It is from the official Rocky 8.8 powering my wolf-69 node.

In what version of _fio_?  I don't see that script as part of their [packaging upstream](https://git.rockylinux.org/staging/rpms/fio)

> The issue is with the leap distribution. I tried to use some conditionals without big success.

Which issue exactly?

Reply:
The leap issue, is the missing package containing the pathfix.py script.

Reply:
Hello @brianjmurrell ,

After more investigations it appears that the python script `pathfix.py` should be available in the Leap package `python3-tools`.  However, this script and all most of the files of the directory `/usr/share/doc/packages/python3-core/Tools` are deleted when this last package is installed.  As illustrated in the following trace, these files and directories are skipped when this package is installed:
```
daos_server@49b545190cf6:~> sudo dnf install -v --rpmverbosity=debug python3-tools
Loaded plugins: builddep, changelog, config-manager, copr, debug, debuginfo-install, download, generate_completion_cache, groups-manager, needs-restarting, playground, repoclosure, repodiff, repograph, repomanage, reposync
DNF version: 4.10.0
...
D: Plugin: calling hook tsm_pre in selinux plugin
D: Plugin: calling hook tsm_pre in syslog plugin
D: running pre-transaction scripts
D: computing 431 file fingerprints
D: excluding directory /usr/share/doc/packages/python3-core/Tools/unittestgui/
...
D: excluding directory /usr/share/doc/packages/python3-core/Tools/ccbench/
D: opening  db index       /usr/lib/sysimage/rpm/Basenames nofsync mode=0x42
...
D: Plugin: calling hook fsm_file_prepare in selinux plugin
D: skipnstate 040755  1 (   0,   0)     0 /usr/share/doc/packages/python3-core/Tools/ccbench
...
D: skipnstate 100644  1 (   0,   0)  9846 /usr/share/doc/packages/python3-core/Tools/scripts/patchcheck.py;65c37f6b
D: skipnstate 100644  1 (   0,   0)  5203 /usr/share/doc/packages/python3-core/Tools/scripts/pathfix.py;65c37f6b
D: skipnstate 100644  1 (   0,   0)  3910 /usr/share/doc/packages/python3-core/Tools/scripts/pdeps.py;65c37f6b
...
D: skipnstate 100644  1 (   0,   0) 18556 /usr/share/doc/packages/python3-core/Tools/unittestgui/unittestgui.py;65c37f6b
D: create     100644  2 (   0,   0)     0 /usr/lib64/python3.6/turtledemo/__pycache__/__init__.cpython-36.opt-1.pyc;65c37f6b
...
```

Sadly, I was not able to find why these files are excluded, thus I will use some workaround to deal with issue.

Reply:
> * [x]  Use pathfix.py from package distrib

Fixed with commit 7430be2ec301f775e4144e5973f74155393b6ca3

Reply:
> Hello @brianjmurrell ,

Hi @knard-intel.
 
> After more investigations it appears that the python script `pathfix.py` should be available in the Leap package `python3-tools`.

Nice find!

> However, this script and most of the files of the directory `/usr/share/doc/packages/python3-core/Tools` are deleted when this last package is installed.

Which last package?

> As illustrated in the following trace, these files and directories are skipped when this package is installed:

Do you have some configuration on your machine that is causing that?  My Leap 15.4 machine here has that file after installing the package:

```
$ ls -l /usr/share/doc/packages/python3-core/Tools/scripts/pathfix.py
-rw-r--r--. 1 root root 5203 Sep 13 14:47 /usr/share/doc/packages/python3-core/Tools/scripts/pathfix.py
```